practical/groovy/analyse_dataset_and_save_rois.groovy

```diff
@@ -0,0 +1,194 @@
+ */
+
+/*
+ * This Groovy script shows how to analyse an OMERO dataset
+ * i.e. a collection of OMERO images.
+ * For that example, we use the analyse particles plugin.
+ * The generated ROIs are then saved back to OMERO.
+ * Use this script in the Scripting Dialog of Fiji (File > New > Script).
+ * Select Groovy as language in the Scripting Dialog.
+ * Error handling is omitted to ease the reading of the script but this
+ * should be added if used in production to make sure the services are closed
+ * Information can be found at
+ * https://docs.openmicroscopy.org/latest/omero5/developers/Java.html
+ */
+
+import java.util.ArrayList
+
+
+// OMERO Dependencies
+import omero.gateway.Gateway
+import omero.gateway.LoginCredentials
+import omero.gateway.SecurityContext
+import omero.gateway.facility.BrowseFacility
+import omero.gateway.facility.ROIFacility
+import omero.log.SimpleLogger
+
+import org.openmicroscopy.shoola.util.roi.io.ROIReader
+
+import loci.formats.FormatTools
+import loci.formats.ImageTools
+import loci.common.DataTools
+
+import ij.IJ
+import ij.ImagePlus
+import ij.ImageStack
+import ij.process.ByteProcessor
+import ij.process.ShortProcessor
+import ij.plugin.frame.RoiManager
+import ij.measure.ResultsTable
+
+
+// Setup
+// =====
+
+// OMERO Server details
+HOST = "outreach.openmicroscopy.org"
+PORT = 4064
+
+//  parameters to edit
+dataset_id = 2331
+USERNAME = "username"
+PASSWORD = "password"
+
+def connect_to_omero() {
+    "Connect to OMERO"
+
+    credentials = new LoginCredentials()
+    credentials.getServer().setHostname(HOST)
+    credentials.getServer().setPort(PORT)
+    credentials.getUser().setUsername(USERNAME.trim())
+    credentials.getUser().setPassword(PASSWORD.trim())
+    simpleLogger = new SimpleLogger()
+    gateway = new Gateway(simpleLogger)
+    gateway.connect(credentials)
+    return gateway
+
+}
+
+def get_image_ids(gateway, ctx, dataset_id) {
+    "List all image's ids contained in a Dataset"
+
+    browse = gateway.getFacility(BrowseFacility)
+
+    ids = new ArrayList(1)
+    ids.add(new Long(dataset_id))
+    images = browse.getImagesForDatasets(ctx, ids)
+
+    j = images.iterator()
+    image_ids = new ArrayList()
+    while (j.hasNext()) {
+        image_ids.add(j.next().getId())
+    }
+    return image_ids
+}
+
+
+def open_image_plus(HOST, USERNAME, PASSWORD, PORT, group_id, image_id) {
+    "Open the image using the Bio-Formats Importer"
+
+    StringBuffer options = new StringBuffer()
+    options.append("location=[OMERO] open=[omero:server=")
+    options.append(HOST)
+    options.append("\nuser=")
+    options.append(USERNAME.trim())
+    options.append("\nport=")
+    options.append(PORT)
+    options.append("\npass=")
+    options.append(PASSWORD.trim())
+    options.append("\ngroupID=")
+    options.append(group_id)
+    options.append("\niid=")
+    options.append(image_id)
+    options.append("] ")
+    options.append("windowless=true")
```

Comment:
we still need the additional space here to match the other script
(i.e. replace "windowless=true"  with  "wndowless=true ")